### PR TITLE
Use secret in appregistry

### PR DESCRIPTION
**Pull request by kevinrizza**
-Update operatorsource spec to include secret ref
-Update appregistry to pass token in auth header
-Update downloading reconciler to pass secret to appregistry
-Update poller to pass secret to appregistry
-Query for secret when setting up appregistry instance

### CODE REVIEW

File: pkg/apis/marketplace/v1alpha1/operatorsource_types.go [hunk 1]
```diff
@@ -13,19 +13,6 @@ const (
...
-
-// OperatorSource is the Schema for the operatorsources API
-// +k8s:openapi-gen=true
-type OperatorSource struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              OperatorSourceSpec   `json:"spec,omitempty"`
-	Status            OperatorSourceStatus `json:"status,omitempty"`
-}
-
```
Comment (tkashem) [inline pkg/apis/marketplace/v1alpha1/operatorsource_types.go]: Same question: did you have to move it to get `operatorsdk k8s generate` to work? I don't see the `// +genclient` tag that was added.

File: pkg/operatorsource/registryhelper.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		key := k8sclient.ObjectKey{
+			Name:      auth.SecretName,
+			Namespace: auth.SecretNamespace,
+		}
+		err := client.Get(context.TODO(), key, &secret)
+		if err != nil {
+			return options, err
+		}
+
+		options.AuthToken = string(secret.Data["token"])
```
Comment (tkashem) [inline pkg/operatorsource/registryhelper.go]: The auth token is already opaque, wondering whether we need to base64 encode it before we store it in the Secret object.

File: pkg/apis/marketplace/v1alpha1/operatorsource_types.go (hunk 1 above)
Comment (kevinrizza) [reply pkg/apis/marketplace/v1alpha1/operatorsource_types.go]: @tkashem I didn't use the sdk to generate it, I just pulled your commit down. I can undo this and reapply it using the sdk if needed.

File: pkg/operatorsource/registryhelper.go (hunk 2 above)
Comment (kevinrizza) [reply pkg/operatorsource/registryhelper.go]: @tkashem Is there a way to store it as a simple string without the encoding? I'm not too familiar with kube secrets, but my understanding is that the only real user data supported type is Opaque. Are you saying we should use the 'StringData' field to save the token in plain text? Won't that be harder to define later?

**Comment by kevinrizza** [comment]
This was replaced by changes in another pr. Closing as obsolete.

**Pull request closed without merging**